ui/src/component/funnel.js

```diff
@@ -0,0 +1,52 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactEcharts from 'echarts-for-react';
+import _ from 'lodash';
+import { validate } from '../utils';
+
+export default class Funnel extends PureComponent {
+  static propTypes = {
+    source: PropTypes.arrayOf(PropTypes.array).isRequired,
+  }
+
+  render() {
+    const { source } = this.props;
+    validate(this.props.source);
+
+    const [name, value] = _.first(source);
+    const keyValueSource = _.map(source.slice(1), item => _.zipObject([name, value], item));
+    // sort the data descending first so that legends could start from highest to lowest
+    const sortedSource = _.reverse(_.sortBy(keyValueSource, value));
```

Comment:
It does not make much sense because sorting is not always needed.
Take this as an example 
 ![chart](https://pic4.zhimg.com/80/v2-56886ed9e9ce23b37f149e5ea206c0ec_hd.jpg)

Reply:
We don't need to implement all kinds of funnels. 
Funnels in wikipedia:
https://en.wikipedia.org/wiki/Funnel_chart

A funnel chart displays values as progressively decreasing proportions amounting to 100 percent in total. The size of the area is determined by the series value as a percentage of the total of all values. Any funnel consists of the higher part called head (or base) and the lower part referred to as neck.